### PR TITLE
feat: set default message queue key to format plain

### PR DESCRIPTION
**Pull request by keven-huang**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Set default message queue key to format plain

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

File: src/connector/src/parser/avro/parser.rs [hunk 1]
```diff
@@ -332,11 +338,12 @@ mod test {
         let flush = writer.flush().unwrap();
         assert!(flush > 0);
         let input_data = writer.into_inner().unwrap();
+        let key_data = vec![];
         let columns = build_rw_columns();
         let mut builder = SourceStreamChunkBuilder::with_capacity(columns, 1);
         {
             let writer = builder.row_writer();
-            parser.parse_inner(input_data, writer).await.unwrap();
+            parser.parse_inner(Some(key_data), Some(input_data), writer).await.unwrap();
```
Comment (tabVersion) [inline src/connector/src/parser/avro/parser.rs:346]: please check if the hidden column `_rw_key` is NULL

File: src/connector/src/parser/plain_parser.rs [hunk 2]
```diff
@@ -58,12 +67,23 @@ impl PlainParser {
 
     pub async fn parse_inner(
         &mut self,
-        payload: Vec<u8>,
+        key: Option<Vec<u8>>,
+        payload: Option<Vec<u8>>,
```
Comment (tabVersion) [inline src/connector/src/parser/plain_parser.rs:71]: why changing to `Option`

File: src/connector/src/parser/unified/util.rs [hunk 3]
```diff
@@ -46,6 +47,19 @@ pub fn apply_row_accessor_on_stream_chunk_writer(
     writer.insert(|column| accessor.access(&[&column.name], Some(&column.data_type)))
 }
 
+pub fn apply_key_val_accessor_on_stream_chunk_writer(
+    key_accessor: AccessImpl<'_, '_>,
+    val_accessor: AccessImpl<'_, '_>,
+    writer: &mut SourceStreamChunkRowWriter<'_>,
+) -> AccessResult<()> {
+    let mut row_op: UpsertChangeEvent<AccessImpl<'_, '_>, AccessImpl<'_, '_>> =
+        UpsertChangeEvent::default();
```
Comment (tabVersion) [inline src/connector/src/parser/unified/util.rs:56]: I dont quite get what this func do here

File: src/connector/src/parser/plain_parser.rs (hunk 2 above)
Comment (keven-huang) [reply src/connector/src/parser/plain_parser.rs:71]: cause if we want to parse key, we need to call `parse_one` and then call `parse inner`. The argument pass by `parse_one` is Option. Previously we don't need to parse key so we just get item inside the `Option<Payload>` in the function `parse_one` by macro. Currently we need to get item both `Option<Payload>` and `Option<Key>`. So I contend that getting item inside the option can be moved into function `parse_inner`

**Review by tabVersion — APPROVED** [review]
LGTM, thanks for your quick support.
please wait for @st1page's review

**Comment by tabVersion** [comment]
also, please remember to update the planner test

**Comment by st1page** [comment]
> also, please remember to update the planner test

Yes, you can refer https://github.com/risingwavelabs/risingwave/blob/main/src/frontend/planner_test/README.md

**Review by st1page — APPROVED** [review]

**Comment by st1page** [comment]
please add some test that define the `Primary key(_rw_key)` . thanks!

File: e2e_test/source/basic/kafka.slt [hunk 4]
```diff
@@ -363,6 +363,23 @@ create table s27 with (
   scan.startup.mode = 'earliest'
 ) FORMAT PLAIN ENCODE JSON (schema.location = 'file:///risingwave/json-complex-schema')
 
+# currently _rw_key can be set as primary key
+statement ok
+create source s28 (id bytea, PRIMARY KEY(_rw_key)) with (
```
Comment (st1page) [inline e2e_test/source/basic/kafka.slt:368]: oh, sorry for my vague description, in RisingWave, creating source with the primary key is forbidden. I just talked about the table. I think you can remove this source test.

Comment (keven-huang) [reply e2e_test/source/basic/kafka.slt:368]: got it

**Pull request merged**